model/src/test/kotlin/DependencyTreeNavigatorTest.kt

```diff
@@ -0,0 +1,179 @@
+                    Identifier("Maven:org.scalacheck:scalacheck_2.12:1.13.5"),
+                    Identifier("Maven:org.scalatest:scalatest_2.12:3.0.4")
+                )
+            }
+
+            "return an empty sequence for an unknown project ID" {
+                val navigator = OrtResult.EMPTY.dependencyNavigator()
+
+                navigator.directDependencies(PROJECT_ID, "someScope").toList() should beEmpty()
+            }
+
+            "return an empty sequence for an unknown scope" {
+                val navigator = testResult.dependencyNavigator()
+
+                navigator.directDependencies(PROJECT_ID, "unknownScope").toList() should beEmpty()
+            }
+        }
+
+        "scopeDependencies" should {
+            "return a map with scopes and their dependencies for a project" {
+                val project = testResult.getProject(PROJECT_ID)!!
+
+                val navigator = testResult.dependencyNavigator()
+
+                val scopeDependencies = navigator.scopeDependencies(project)
+
+                scopeDependencies.keys should containExactlyInAnyOrder("compile", "test")
+
+                val compileDependencies = scopeDependencies["compile"].orEmpty()
+                compileDependencies should haveSize(17)
+                compileDependencies should containAll(
+                    Identifier("Maven:com.typesafe.akka:akka-actor_2.12:2.5.6"),
+                    Identifier("Maven:org.scala-lang:scala-reflect:2.12.2"),
+                    Identifier("Maven:org.scala-lang:scala-library:2.12.3")
+                )
+
+                val testDependencies = scopeDependencies["test"].orEmpty()
+                testDependencies should haveSize(6)
+                testDependencies should containAll(
+                    Identifier("Maven:org.scalacheck:scalacheck_2.12:1.13.5"),
+                    Identifier("Maven:org.scalactic:scalactic_2.12:3.0.4")
+                )
+            }
+
+            "return a map with scopes and their dependencies for a project ID" {
+                val navigator = testResult.dependencyNavigator()
+
+                val scopeDependencies = navigator.scopeDependencies(PROJECT_ID)
+
+                scopeDependencies.keys should containExactlyInAnyOrder("compile", "test")
+
+                scopeDependencies["compile"].orEmpty() should haveSize(17)
+                scopeDependencies["test"].orEmpty() should haveSize(6)
+            }
+
+            "return a map with scopes and their dependencies for an unknown project ID" {
+                val navigator = OrtResult.EMPTY.dependencyNavigator()
+
+                navigator.scopeDependencies(Identifier.EMPTY).keys should beEmpty()
+            }
+
+            "return a map with scopes and their dependencies up to a given maxDepth" {
+                val navigator = testResult.dependencyNavigator()
+
+                val scopeDependencies = navigator.scopeDependencies(PROJECT_ID, maxDepth = 1)
```

Comment:
I think it would be good to also test with another depth, especially as 1 is the simplest case.